src/cpu/dyn_cache.h

```diff
@@ -739,13 +739,19 @@ static constexpr size_t cache_blocks_total_bytes = CACHE_BLOCKS * sizeof(CacheBl
 
 static inline void dyn_mem_execute(MAYBE_UNUSED void* ptr, MAYBE_UNUSED size_t size)
 {
+	// Align to page boundary and adjust size
+	const auto p = reinterpret_cast<uintptr_t>(ptr) - 1;
+	const auto align_adjust = p % PAGESIZE_TEMP;
+	const auto p_aligned = p - align_adjust;
+	size += align_adjust + 1;
+	ptr = reinterpret_cast<void*>(p_aligned);
+
```

Comment:
With the two alignment blocks of the same code, maybe lift it out -

``` c++
// Align to page boundary and adjust size
static void dyn_mem_align(void* & ptr, size_t & size)
{	const auto p = reinterpret_cast<uintptr_t>(ptr) - 1;
	const auto align_adjust = p % PAGESIZE_TEMP;
	const auto p_aligned = p - align_adjust;
	size += align_adjust + 1;
	ptr = reinterpret_cast<void*>(p_aligned);
}
```

Reply:
I like it. Done.